src/definition/populate.go

```diff
@@ -116,7 +116,12 @@ func IntegrationPopulator(config *IntegrationPopulateConfig) (bool, []error) {
 				extraAttributes...,
 			)
 
-			msType, err := config.MsTypeGuesser(config.ClusterName, groupLabel, entityID, config.Groups)
+			msTypeGuesser := config.MsTypeGuesser
+			if customGuesser := config.Specs[groupLabel].MsTypeGuesser; customGuesser != nil {
```

Comment:
I still do not understand why instead of having `config.Specs[groupLabel].MsTypeGuesser` we do not have `config.Specs[groupLabel].MsType` directly 😅 don't you have the feeling that there is too much code to decide a sample name?

Reply:
offline but online 😄